src/pyk/proof/reachability.py

```diff
@@ -80,6 +80,7 @@ class APRProof(Proof, KCFGExploration):
     circularity: bool
     _exec_time: float
     error_info: Exception | None
+    prior_loops_cache: dict[NodeIdLike, list[NodeIdLike]]
```

Comment:
```suggestion
    prior_loops_cache: dict[int, list[int]]
```

Please make the necessary adjustments so that we're not storing node aliases or node strings here, just integers. That means, when updating the `prior_loops_cache`, we need to use `KCFG.resolve(node_id)`, to get out the actual integer referring to the node.

Reply:
Yeah, if we already have the `id` from `node.id`, we don't need `KCFG.resolve(...)`.